v20.2/sso.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Single Sign-on (Enterprise)
+summary: Implement single sign-on (SSO) for Admin UI access.
+toc: true
+---
+
+Single sign-on (SSO) allows a user to access the Admin UI in a secure cluster via an external identity provider. When SSO is configured and enabled, the [Admin UI login page](admin-ui-overview.html#admin-ui-access) will display an OAuth login button in addition to the password access option.
+
+CockroachDB supports SSO via [OpenID Connect (OIDC)](https://openid.net/connect/), an authentication layer built on top of OAuth 2.0.
+
+{{site.data.alerts.callout_info}}
+SSO authentication is an [enterprise-only](enterprise-licensing.html) feature.
+{{site.data.alerts.end}}
+
+{% include {{ page.version.version }}/misc/experimental-warning.md %}
+
+## Requirements
+
+- An external OAuth 2.0 identity provider
+
+## Login flow
+
+This SSO implementation uses the [authorization code grant type](https://tools.ietf.org/html/rfc6749#section-4.1) to authenticate a user.
+
+1. A user opens the Admin UI and clicks on the OAuth login button. 
+1. The user is redirected to an external identity provider.
+1. The user successfully authenticates with the provider, completing the OAuth flow.
+1. The user is redirected to the CockroachDB cluster via a callback URL.
+1. The authorization code in the callback query is exchanged for an [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken).
+1. CockroachDB matches the ID Token to a registered SQL user.
+1. CockroachDB creates a web session for the SQL user.
+1. The user is redirected to the [Admin UI Cluster Overview](admin-ui-cluster-overview-page.html).
+
+## Cluster settings
+
+The following OIDC [cluster settings](cluster-settings.html) are used to configure and enable SSO.
+
+| Cluster Setting | Example Value
+|-----------------|------
+| `server.oidc_authentication.enabled` | true
+| `server.oidc_authentication.client_id` | '32789079457-g3hdfw8cbw85obi5cb525hsceaqf69unn.apps.googleusercontent.com'
+| `server.oidc_authentication.client_secret` | '6Q_jmRMgrPNOc_mN91boe-9EP'
+| `server.oidc_authentication.redirect_url` | 'https://localhost:8080/oidc/v1/callback'
+| `server.oidc_authentication.provider_url` | 'https://accounts.google.com'
+| `server.oidc_authentication.scopes` | 'openid profile email'
+| `server.oidc_authentication.claim_json_key` | 'email'
+| `server.oidc_authentication.principal_regex` | '^([^@]+)@[^@]+$'
+
+- `enabled` is a Boolean that enables or disables SSO.
```

Comment:
We could add another column named "Description" and include these bullet points in the table.